ios/browser/api/sync/brave_sync_internals.mm

```diff
@@ -0,0 +1,217 @@
+}
+
+}  // namespace web
+
+@protocol CRWebViewPrivate<NSObject>
+- (void)setWebView:(WKWebView*)webView;
+@end
+
+@interface BraveSyncInternalsView ()<CRWWebStateDelegate, CRWWebStateObserver> {
+  ChromeBrowserState* _chromeBrowserState;
+  std::unique_ptr<web::WebState> _webState;
+  std::unique_ptr<web::WebStateObserverBridge> _webStateObserver;
+  std::unique_ptr<web::WebStateDelegateBridge> _webStateDelegate;
+  
+  WKWebView* _webView;
+  CRWWebController* _webController;
+}
+@end
+
+@implementation BraveSyncInternalsView
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)mainBrowserState {
+  if ((self = [super init])) {
+    _chromeBrowserState = mainBrowserState;
+    web::WebState::CreateParams webStateCreateParams(mainBrowserState);
+    _webState = web::WebState::Create(webStateCreateParams);
+    //web::WebState::CreateWithStorageSession(webStateCreateParams,
+    //                                        sessionStorage);
+    
+    _webStateObserver = std::make_unique<web::WebStateObserverBridge>(self);
+    _webState->AddObserver(_webStateObserver.get());
+
+    _webStateDelegate = std::make_unique<web::WebStateDelegateBridge>(self);
+    _webState->SetDelegate(_webStateDelegate.get());
+    
+//    web::WKWebViewConfigurationProvider& config_provider =
+//          web::WKWebViewConfigurationProvider::FromBrowserState(_chromeBrowserState);
+//
+//    _webView = web::EnsureWebViewCreatedWithConfiguration(_webState.get(),
+
+    _webController = static_cast<web::WebStateImpl*>(_webState.get())->GetWebController();
+    _webView = [_webController ensureWebViewCreated];
+    
+    UIView* view = [_webController view];
+    [self addSubview:view];
+    [view setTranslatesAutoresizingMaskIntoConstraints:NO];
+    
+    NSLayoutConstraint *leading = [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeLeading multiplier:1.0 constant:0.0];
+    
+    NSLayoutConstraint *trailing = [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeTrailing multiplier:1.0 constant:0.0];
+    
+    NSLayoutConstraint *top = [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0];
+    
+    NSLayoutConstraint *bottom = [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+
+    NSArray* constraints = @[leading, trailing, top, bottom];
+    [self addConstraints:constraints];
+    [NSLayoutConstraint activateConstraints:constraints];
+  }
+  return self;
+}
+
+- (void)dealloc {
+  [[_webController view] removeFromSuperview];
+  _webController = nil;
+  _webView = nil;
```

Comment:
Not necessary with ARC

Reply:
It is if you want a specific order of destruction, which is what I want. I want to make sure things are cleaned up in the exact order presented here.